library/std/src/net/udp.rs

```diff
@@ -157,7 +167,8 @@ impl UdpSocket {
     }
 
     /// Sends data on the socket to the given address. On success, returns the
-    /// number of bytes written.
+    /// number of bytes written. Note that partial writes are not possible with
+    /// UDP.
```

Comment:
And it may seem unreasonable to hinge this on some pedantry: who the fuck is trying to push 2GiB buffers over UDP? But a network stack can do packet-splitting for UDP and thus accept and try to transmit huge buffers like that, and if someone is relying on "no partial writes" but they send, say, 3GiB, well... yeah.

And nowadays we have like 10Gb Ethernet, even for (higher-end) client hardware, so.

Reply:
This is fascinating and weird. Without data to back it up, I wouldn't expect any software to actually send packages large enough to be split up **and** handle the returned `usize`.

According to RFC 791 page 12, the size of a UDP packet is a 16-bit number, suggesting that packets larger than 2^16 (minus one) are impossible. There doesn't seem to be an extension to solve this. In IPv6, there are apparently jumbograms, but that seems to me like a whole other topic.

So I'm not sure if it is actually possible to send a UDP packet larger than 2^16, which then in turn would make partial write impossible.

Reply:
I am definitely not sure about anything currently, but maybe this part of the documentation (about the maximum package size, partial writes) should be at the top of the page with a link to the RFC, not mentioned at each `send` function variant.
Does that sound good?

Reply:
I _think_ jumbograms de facto allow sending larger UDP packets? People do a lot of slightly-off-spec stuff in networking yet somehow it works out.

But yes, I am happy to have us properly link stuff instead. That makes it less likely this documentation falls afoul of my main concern: that a later spec could make it formally untrue, but no one can figure out it's wrong, so it remains taken as an authority.